### ISSUE TITLE
Facets not loaded when not displayed at opening page

### ISSUE THREAD

**Issue by camilstaps** [simile-widgets/exhibit#167]
Please take a look at the following page (source code below): http://exhibit.camilstaps.nl/display-none.html.

I have my facets in a container with `display:none;` to show them on a later moment. When I click 'toggle facets', the facets are shown but are empty. Is that by design / how do I resolve this?

---

This is the relevant HTML. I'm using the data of the [senate example](http://www.simile-widgets.org/exhibit3/examples/senate/senate.html).

```html
<div data-ex-role="collection" data-ex-item-types="Senator"></div>
<table>
	<tr>
		<td>
			<button id="toggle-facets">Toggle facets</button>
			<div id="facet-list" style="display:none;">
				<div data-ex-role="facet"
				     data-ex-expression=".party"></div>
				<div data-ex-role="facet"
				     data-ex-expression=".state"></div>
			</div>
		</td>
		<td>
			<div data-ex-role="viewPanel">
				<div data-ex-role="view"
				     data-ex-view-class="Tabular"
				     data-ex-columns=".label, .party, .state"
				     data-ex-columns="name, party, state"></div>
			</div>
		</td>
	</tr>
</table>
```

**Comment by karger** [comment]
You're running into a conflict with exhibit's collapsible facet 
functionality. If you add a data-ex-collapsible argument to your facet, 
it becomes one that can be collapsed (there's also a data-ex-collapsed 
argument that tells it to start in collapsed from).  Collapsed facets 
don't get updated, which saves computation time; conversely, expanding 
the facet triggers it to compute itself.

Unfortunately, the way exhibit *checks* whether a facet is collapsed is 
by checking whether it is visible.  So starting your facet invisible 
tells exhibit not to bother updating it. Unfortunately making the facet 
visible is *not* the same as uncollapsing the facet, so exhibit doesn't 
notice the change and doesn't update the facet.

if using the collapse arguments solves your problem, great.  if not, I'm 
unsure when we can find time to try addressing this issue in the exhibit 
code.

**Comment by camilstaps** [comment]
Thanks for the quick reply!

In fact, this was just a minimal example, the actual file is a bit larger and I don't think `data-ex-collapsed` could really work there.

Would it perhaps be possible to add something like `data-ex-force-load` to forcefully override the check you are talking about?

Or: is there a way that I can tell Exhibit from *my* JavaScript that some facets should be loaded? Then I can add that to the toggle function.

**Comment by karger** [comment]
it isn't just a matter of loading.  the facets need to be paying 
attention and updating each time any other filter on the page is 
modified.   if exhibit thinks the facet is collapsed none of that will 
happen.  I'm to sleepy right now to be able to think of a workaround 
that doesn't involve programming; I'll try again tomorrow.

**Comment by camilstaps** [comment]
Any news on this?

I thought I can get the same result using `z-index` and absolute positioning, but something like `data-ex-force-refresh` would be a much neater way from the programmer's point of view.

**Comment by averynortonsmith** [comment]
Hi  camilstaps

I'm working on this bug, should have it fixed by tomorrow.

- Avery

**Comment by averynortonsmith** [comment]
I believe this issue is fixed. See [http://web.mit.edu/averyn/www/test/exhibit/scripted/demos/test/facets.html](http://web.mit.edu/averyn/www/test/exhibit/scripted/demos/test/facets.html) for a demo. 

Modified the isCollapsed function to only identify facets that are hidden AND collapsible as being collapsed.

Opening a pull request with the updated code

**Comment by camilstaps** [comment]
@averynortonsmith, I will try it out later today. Perhaps it's better to open two PRs though, so that they can be merged separately.

**Comment by karger** [comment]
looks like the current pr might break google maps.   we'll work on 
separating the PRs

**Comment by camilstaps** [comment]
Is there now still any reason to not merge #173 (and #174)? If there is a problem with them, perhaps I can help.

**Comment by karger** [comment]
The PR fixes your specific problem but we've realized that it leaves in some problems for other cases.  We've got a better fix planned that will solve the problem generally.  If you're able to work off a local copy you can merge in the PR yourself.  If you're in a bind and need it deployed fast we can do that; it'll just mess up the commit history a bit.

**Comment by camilstaps** [comment]
Thanks David, that explains. I don't have to deploy this code yet, so no hurry is needed. I'm working off a local copy now, as you suggested, and can continue my work until the general fix is released. Thanks!

**Comment by camilstaps** [comment]
Fixed in #173. Thanks!
